modules/programs/firefox/mkFirefoxModule.nix

```
@@ -666,37 +662,106 @@ in {
-              List of ${name} add-on packages to install for this profile.
-              Some pre-packaged add-ons are accessible from the
-              [Nix User Repository](https://github.com/nix-community/NUR).
-              Once you have NUR installed run
+            type = types.coercedTo (types.listOf types.package) (packages: {
+              packages = mkIf (builtins.length packages > 0) (warn ''
+                In order to support declarative extension configuration,
+                extension installation has been moved from
+                programs.firefox.profiles.<profile>.extensions
+                to
+                programs.firefox.profiles.<profile>.extensions.packages
+              '' packages);
+            }) (types.submodule {
+              options = {
+                packages = mkOption {
+                  type = types.listOf types.package;
+                  default = [ ];
+                  example = literalExpression ''
+                    with pkgs.nur.repos.rycee.firefox-addons; [
+                      privacy-badger
+                    ]
+                  '';
+                  description = ''
+                    List of ${name} add-on packages to install for this profile.
+                    Some pre-packaged add-ons are accessible from the Nix User Repository.
+                    Once you have NUR installed run
 
-              ```console
-              $ nix-env -f '<nixpkgs>' -qaP -A nur.repos.rycee.firefox-addons
-              ```
+                    ```console
+                    $ nix-env -f '<nixpkgs>' -qaP -A nur.repos.rycee.firefox-addons
+                    ```
 
-              to list the available ${name} add-ons.
+                    to list the available ${name} add-ons.
 
-              Note that it is necessary to manually enable these extensions
-              inside ${name} after the first installation.
+                    Note that it is necessary to manually enable these extensions
+                    inside ${name} after the first installation.
+
+                    To automatically enable extensions add
+                    `"extensions.autoDisableScopes" = 0;`
+                    to
+                    [{option}`${moduleName}.profiles.<profile>.settings`](#opt-${moduleName}.profiles._name_.settings)
+                  '';
+                };
 
-              To automatically enable extensions add
-              `"extensions.autoDisableScopes" = 0;`
-              to
-              [{option}`${moduleName}.profiles.<profile>.settings`](#opt-${moduleName}.profiles._name_.settings)
+                settings = mkOption {
+                  default = { };
+                  description = ''
+                    ExtensionID's for addons that you wish to configure.
+
+                    This can be in the form of either an ID string
+                    ```nix
+                    # extensionID for uBlock Origin
+                    "uBlock0@raymondhill.net".options = {
```

Judging from the test you added I think you renamed that option to `settings`. https://github.com/nix-community/home-manager/blob/678dd3daeb85ee39d37561dc73aff73c1b1cfec4/tests/modules/programs/firefox/profiles/extensions/default.nix#L14
```suggestion
                    "uBlock0@raymondhill.net".settings = {
```